### DISCUSSION TITLE
Optimization and 2 more

### DISCUSSION THREAD

**Discussion by YstLol** [stakira/OpenUtau#596] [Ideas]
### Optimization
There is an **FPS difference** between maximized piano roll and restored piano roll.
If the window gets smaller, it gains FPS. It's opposite if you make the window bigger.
### Highlighting playing notes like UTAU
UTAU highlights the playing note, while OpenUTAU doesn't.
![image](https://user-images.githubusercontent.com/98826870/218757466-c6e2bc70-db28-41b4-babe-3084cb63b63d.png)
![image](https://user-images.githubusercontent.com/98826870/218757931-cf340fff-51db-4c8d-a6ed-b1aa163af479.png)
### Maybe an MP4 generator?
If you can't improve the FPS, why not render an MP4? Some people make videos with screen recorders, but it's the same FPS. It can be like Utauview. But better. OpenUTAU UI, singer name, 60 fps, options what to view.

**Comment by oxygen-dioxide** [comment]
I don't think we should make a video exporter builtin to OpenUtau. It wouldn't be easier to implement than a independent video maker from ustx file. A separate software or plugin is OK. Video and audio editor softwares like Audacity, Adobe Premiere, fl studio (or any other daw) don't provide "Export UI as video" feature. 
If you feel the UI laggy in screen records, turn "Auto-Scroll" to "Page scroll" in preferences.

**Reply by oxygen-dioxide** [reply]
Video maker isn't in the goal of OpenUtau. You can also take a look at these open source projects: 
pypianoroll: https://github.com/salu133445/pypianoroll
svg plugin for opensvip converter: https://github.com/yqzhishen/opensvip/tree/main/csharp/Plugins/Svg
Euphony: https://github.com/qiao/euphony
